osquery/filesystem/windows/fileops.cpp

```diff
@@ -1469,6 +1473,245 @@ LONGLONG filetimeToUnixtime(const FILETIME& ft) {
   return date.QuadPart / 10000000;
 }
 
+LONGLONG longIntToUnixtime(LARGE_INTEGER& ft) {
+  ULARGE_INTEGER ull;
+  ull.LowPart = ft.LowPart;
+  ull.HighPart = ft.HighPart;
+  return ull.QuadPart / 10000000ULL - 11644473600ULL;
+}
+
+std::string getFileAttribStr(unsigned long file_attributes) {
+  std::string attribs;
+
+  if (file_attributes & FILE_ATTRIBUTE_ARCHIVE) {
+    // Archive file attribute
+    attribs.push_back('A');
+    file_attributes &= ~FILE_ATTRIBUTE_ARCHIVE;
+  }
+  if (file_attributes & FILE_ATTRIBUTE_COMPRESSED) {
+    // Compressed (Not included in attrib.exe output)
+    attribs.push_back('C');
+    file_attributes &= ~FILE_ATTRIBUTE_COMPRESSED;
+  }
+  if (file_attributes & FILE_ATTRIBUTE_ENCRYPTED) {
+    // Encrypted (Not included in attrib.exe output)
+    attribs.push_back('E');
+    file_attributes &= ~FILE_ATTRIBUTE_ENCRYPTED;
+  }
+  if (file_attributes & FILE_ATTRIBUTE_REPARSE_POINT) {
+    // Hidden file attribute
+    attribs.push_back('L');
+    file_attributes &= ~FILE_ATTRIBUTE_HIDDEN;
+  }
+  if (file_attributes & FILE_ATTRIBUTE_HIDDEN) {
+    // Hidden file attribute
+    attribs.push_back('H');
+    file_attributes &= ~FILE_ATTRIBUTE_HIDDEN;
+  }
+  if (file_attributes & FILE_ATTRIBUTE_INTEGRITY_STREAM) {
+    //
+    attribs.push_back('V');
+    file_attributes &= ~FILE_ATTRIBUTE_INTEGRITY_STREAM;
+  }
+  if (file_attributes & FILE_ATTRIBUTE_NORMAL) {
+    // Normal (Not included in attrib.exe output)
+    attribs.push_back('N');
+    file_attributes &= ~FILE_ATTRIBUTE_NORMAL;
+  }
+  if (file_attributes & FILE_ATTRIBUTE_NOT_CONTENT_INDEXED) {
+    // Not content indexed file attribute
+    attribs.push_back('I');
+    file_attributes &= ~FILE_ATTRIBUTE_NOT_CONTENT_INDEXED;
+  }
+  if (file_attributes & FILE_ATTRIBUTE_NO_SCRUB_DATA) {
+    // No scrub file attribute
+    attribs.push_back('X');
+    file_attributes &= ~FILE_ATTRIBUTE_NO_SCRUB_DATA;
+  }
+  if (file_attributes & FILE_ATTRIBUTE_OFFLINE) {
+    // Offline attribute
+    attribs.push_back('O');
+    file_attributes &= ~FILE_ATTRIBUTE_OFFLINE;
+  }
+  if (file_attributes & FILE_ATTRIBUTE_READONLY) {
+    // Read-only file attribute
+    attribs.push_back('R');
+    file_attributes &= ~FILE_ATTRIBUTE_READONLY;
+  }
+  if (file_attributes & FILE_ATTRIBUTE_SYSTEM) {
+    // System file attribute
+    attribs.push_back('S');
+    file_attributes &= ~FILE_ATTRIBUTE_SYSTEM;
+  }
+  if (file_attributes & FILE_ATTRIBUTE_TEMPORARY) {
+    // Temporary file attribute (Not included in attrib.exe output)
+    attribs.push_back('T');
+    file_attributes &= ~FILE_ATTRIBUTE_TEMPORARY;
+  }
+
+  return attribs;
+}
+
+int platformStat(const fs::path& path, WINDOWS_STAT* wfile_stat) {
```

Comment:
Rather than returning an `int` here, let's change this to be `Status platformStat(const fs::path& path, WINDOWS_STAT* wfile_stat)`, and then the return can be either `return Status(-1, "Platform Stat failed with " + std::to_string(GetLastError()));` or `return Status();` as that value from GLE might come in handy.

Reply:
Done. Actually considered doing this previously. Not sure why I didn't :)